### DISCUSSION TITLE
AgContext feature request: Include additonal options under "NOT IN CROP"

### DISCUSSION THREAD

**Discussion by mkutu** [Weed-AI/Weed-AI#588] [Ideas]
Could you include options like "greenhouse", or "potted plants" as a "NOT IN CROP" option in AgContext?

Doing so would help make it more flexible to diverse datasets while increasing specificity for filtering. Even adding tags like "urban" or "dirt road" would broaden the scope of environments while retaining a weed detection focus.

**Comment by jnothman** [comment]
How does this relate to surface cover / soil colour? do you think they're orthogonal?

also @geezacoleman do you think there would be risks to widening the scope beyond agricultural land contexts?

**Comment by mkutu** [comment]
@jnothman, fair question. In short, not sure. But I could see a couple of situations where specifying unique "Crop type", "soil colour" and "soil coverage" beyond a typical agricultural land context might be useful:

1. Regarding your questions about soil coverage and color; one might be planting in a greenhouse using different soil types
    - potting soil
    - pots with mulch
    - or clay soil from the field
![c6295f23-1602-4907-b4d2-73f99faa2876](https://user-images.githubusercontent.com/54868229/147346286-305b9ebb-3a79-4519-b1e9-42587e83b483.JPG)

![IMG_4225](https://user-images.githubusercontent.com/54868229/147346524-deec5f28-d820-407c-b49c-bba25249a2c4.jpg)

Both of these examples above have been used to collect weed images of simple weed scenes that can be easily annotated for detection in the field. The second picture is not the greatest example but you get the idea.

I know we'll be experimenting with different residue types in our potted plants.

2. Regarding your comment to @geezacoleman about extending beyond agricultural land contexts, we have a dataset of Poa annua (debatably one of most invasive turf grass weeds) that might not fit a typical AgContext option because it was grown in a nursery (below). We used this dataset to create synthetic data for detection in golf courses.
![IMG_4224](https://user-images.githubusercontent.com/54868229/147347139-6320e378-8a50-4755-be09-6d9edfc305c1.jpg)

**Comment by jnothman** [comment]
Would you use a mix of soil types in an industry context? or in an experimental/research context?

**Comment by geezacoleman** [comment]
Speaking from my experience, field scenarios are likely to be consistent soil type and surface coverage generally (obviously some variability between localised areas being the nature of agriculture). The scenarios mentioned above are more edge cases for research and while probably more inconvenient, it would be better to upload individually rather than grouped if they are largely different.

On the different types of surface cover front, the ones we have listed were developed with large-scale grain production systems in mind . More flexibility there would be useful if we're involving datasets from more diverse contexts and groups. So adding categories below might address the gaps:
- turfgrass
- pots/potting soil

Potting soil is a slightly more difficult one, because it is technically surface cover: 'None', soil colour: 'whatever suits', but I think it is important to allow people to select for/against these greenhouse scenarios specifically.

> Regarding your comment to @geezacoleman about extending beyond agricultural land contexts, we have a dataset of Poa annua (debatably one of most invasive turf grass weeds) that might not fit a typical AgContext option because it was grown in a nursery (below). We used this dataset to create synthetic data for detection in golf courses

If I were uploading that dataset, it would be surface cover: None, soil colur:'brown' and then describe the collection conditions in your agcontext.

**Comment by jnothman** [comment]
What's the reln between turfgrass and pasture?

**Comment by geezacoleman** [comment]
Pasture could be a very wide and diverse range of species - e.g. DeepWeeds is rangeland 'pasture'. Pastures often have species like clover, ryegrass, medics etc. Likely evidence of grazing or inconsistent growth. Not typically mown at all.

Turfgrass is a very specific set of grass species that aim to have highly consistent growth stage/patterns and are probably mowed. It's likely to be a singular species and won't have any broadleaf plants (unless they are weeds). If this was a platform developed by turfgrass funding you'd probably have > 5 different species listed but perhaps just 'turfgrass' is the specificity we need. @mkutu thoughts?

**Comment by mkutu** [comment]
> If this was a platform developed by turfgrass funding you'd probably have > 5 different species listed but perhaps just 'turfgrass' is the specificity we need. @mkutu thoughts?

Fair point, I guess I keep forgetting about the focus on row crops. Having said that, I would add another major difference for developing a turf weed detection dataset would be a lack of soil background for "in-field" examples. Instead you'd have different grasses with different colors (winter dormant brown, green, yellowing) and different growth stages (mowed, unmowed, sod, think different types of grass lengths on golf course, football fields, etc.). 

> Would you use a mix of soil types in an industry context?

yes, If our goal is to use same or similar models across different areas (eg. parts of the country), which makes sense. I think one of the big issues is the transferability of models across different agricultural landscapes even within the same cropping system (training on images in one place, and inferencing in many others). Changing soil color, types, textures, residues, moisture, cultural practices will be things that we do to capture more variability that we expect to find in the real world. I think especially on the industry side, capturing edge cases that make up the long tail of a data distribution will be important. I think we could probably upload these individually based on major differences.  

> If I were uploading that dataset, it would be surface cover: None, soil colur:'brown' and then describe the collection conditions in your agcontext.

Makes sense. Got it, thanks.

**Comment by geezacoleman** [comment]
Thought I would move this to the new, fancy Discussions page and give it a whirl.

> Having said that, I would add another major difference for developing a turf weed detection dataset would be a lack of soil background for "in-field" examples. Instead you'd have different grasses with different colors (winter dormant brown, green, yellowing) and different growth stages (mowed, unmowed, sod, think different types of grass lengths on golf course, football fields, etc.).

Interesting, turfgrass really opens up some new cases. I have been reading Jialin's paper's recently on weed detection in turfgrass and these issues have become more clear. I forgot about it before but different turfgrass species would be covered under 'other crop', then type in the scientific name of the species, rather than under the 'not in crop' tab. It sounds like it's more the flexibility of the surface coverage and BBCH scale that's the issue. Without a substantial redesign there to incorporate those specific backgrounds it might be difficult to capture it appropriately.

I guess the workaround would be:

1.  define the 'crop' as say Cynodon dactylon (for bermudagrass)
2. if it's full coverage crop then say soil colour is not_visible
3. Surface cover - none
4. Surface coverage - na
5. Describe the grass conditions in the free text field.

>  Changing soil color, types, textures, residues, moisture, cultural practices will be things that we do to capture more variability that we expect to find in the real world.

Agreed that this is what's needed for singular models to be deployable in a wide range of industry-relevant conditions. I would add though that for people to filter for their missing soil colour/background type etc to train this type of model, the datasets would need to be uploaded separately.

**Comment by jnothman** [comment]
It also reminds me that we could do with a FAQ... or a link to Discussions..?

**Reply by geezacoleman** [reply]
FAQ would be handy, could we have it as an expandable/drop down section on the About page?
